commit 243698ea8552143f33b8c5c272c76ba6cf4be054
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 23:41:15 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/lamarck/patch-no-enforced-universal-on-linux.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 7224e2b8f..9153f889b 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -3101,7 +3101,7 @@ proc macports::_upgrade_mport_deps {mport target} {
                                     ui_error "its dependency $dep_portname is only installed for the arch '$active_archs'"
                                     ui_error "and the configured universal_archs '$universal_archs' are not sufficient."
                                     return -code error "architecture mismatch"
-                                } else {
+                                } elseif {${macports::os_platform} eq "darwin"} {
                                     # upgrade the dep with +universal
                                     dict set variants universal +
                                     dict set options ports_upgrade_enforce-variants yes
